.github/workflows/codecov.yml

```diff
@@ -0,0 +1,98 @@
+name: Prune Codecov .coveragerc
+on: # yamllint disable-line rule:truthy
+  push:
+    paths:
```

Comment:
I initially put these in for testing.

But later I realised it might help to make sure that if someone tinkers with ci.yml or related files, they don't break codecov.yaml without it triggering some kind of test failure. Maybe that is best achieved by having this file filter, but only for the dev branch.

Reply:
I've now updated the triggers to be only on pull requests to dev AND when one of the two files changes.